### PR TITLE
make fields startDate, endDate, memberRate and customerRate optional

### PR DESCRIPTION
**Pull request by imcaizheng**
# Notes
- The weekly cost logic is slightly updated so that any resource booking without `memberRate` are simply ignore when calcuating the weekly cost.
- Also see https://github.com/topcoder-platform/taas-es-processor/pull/6

### CODE REVIEW

**Review by maxceem — CHANGES_REQUESTED** [review]
@imcaizheng the changes look good and I've already merged ES processor changes.

Though for this change we would also need a DB migration script, otherwise we are getting error like:
```
ERROR : null value in column "start_date" violates not-null constraint
```

Let's use Sequalize migrations for this:
- https://sequelize.org/master/manual/migrations.html
- create folder `migrations` where we would put all migrations scripts
- create a migration script to make these fields optional
- create npm commands like:
   ```
   "migrate": "./node_modules/.bin/sequelize db:migrate",
   "migrate:undo": "./node_modules/.bin/sequelize db:migrate:undo",
   ```
   
I would update the budget for this issue to cover this.

File: package.json [hunk 1]
```diff
@@ -11,6 +11,8 @@
     "init-db": "node src/init-db.js",
     "create-index": "node scripts/createIndex.js",
     "delete-index": "node scripts/deleteIndex.js",
+    "migrate": "npx sequelize db:migrate",
+    "migrate:undo": "npx sequelize db:migrate:undo",
```
Comment (imcaizheng) [inline package.json:15]: I tried to run migration and then undo the migration, all is good.
When I tried to run the migration again and create a resource booking without optional fields and then undo the migration, it failed with error message:
```
ERROR: column "start_date" of relation "resource_bookings" contains null values.
```
@maxceem Do you think it is an issue that should be fixed?

Comment (maxceem) [reply package.json:15]: @imcaizheng I think this is fine. As we don't have any default values for these fields, we would have to think how to fill the data first, in such a case.

**Review by maxceem — CHANGES_REQUESTED** [review]
@imcaizheng all works good.

The only one thing we can improve is to use ENV variable for DB configuration instead of hardcoded values, because we cannot add to config real DB passwords.

Let's use `process.env.NEV_VARIABLE_NAME` in config, as suggested in docs https://monosnap.com/file/CsG9bjBfpDKUn6Bfz3Cwskee6e3IQk:
   - `DB_USERNAME`
   - `DB_PASSWORD`
   - `DB_NAME`
   - `DB_HOST`
   
We can use the same variable names for all environments.

**Comment by imcaizheng** [comment]
@maxceem Fixed.

**Review by maxceem — APPROVED** [review]
Thanks, @imcaizheng. All good.

I would wait for Nkumar to come back on Wednesday before merging this PR as I would need his help to run the migration.

**Pull request merged**